### PR TITLE
Added comments

### PR DESCRIPTION
**Pull request by taodario**
Added comments to the code

### CODE REVIEW

**Comment by 123Chan** [comment]
It's really good, just 2 spelling errors and 1 error on bitboard description I noted on the Files changed. Also the extra information that I noted on valid moves should be included.

The Chessboard.java file and Main.java classes will not be used though, but we can keep it for now.

Thank you for doing this, can you just edit the changes I marked and make a pull request again?

**Review by 123Chan — CHANGES_REQUESTED** [review]

File: src/main/java/InitialPositions.java [hunk 1]
```diff
@@ -1,9 +1,14 @@
+/**
+ * This class provides constants for initializing a new chess game.
+ * Each constant represents the initial position of the corresponding piece on a chessboard, with the white pieces at the bottom and the black pieces at the top!
+ * The constants are long values that represent bitboards, with each bit corresponding to a square on the chessboard
+ * The bitboards are read from left to right and top to bottom (top left square is the most significant bit) (i.e., top left hand side of chess board)
```
[the remark this answers is not in the document]
Comment (123Chan) [reply src/main/java/InitialPositions.java:5]: that 1 is h8

[the remark this answers is not in the document]
Comment (123Chan) [reply src/main/java/InitialPositions.java:5]: so rank 8 file h

**Review by 123Chan — CHANGES_REQUESTED** [review]

File: src/main/java/Calculator.java [hunk 2]
```diff
@@ -1,74 +1,34 @@
+/**
+ * This interface is used to calculate valid moves and attack coverage for a chess piece on a chess board.
+ * Used in classes that represent specific types of chess pieces (Pawn, Rook, Knight, Bishop, Queen, King)
+ * The group should be implementing this interface and provide appropriate implementations for the methods.
+ * Note: the Calculator assumes that the player is not in Check. (CheckmateCalculator will be responsible for that.)
```
Comment (123Chan) [inline src/main/java/Calculator.java:5]: Checkmate Calculator will just check for checkmate.

The use case for choosing a piece and displaying valid moves will handle that.

**Review by 123Chan — CHANGES_REQUESTED** [review]

**Review by 123Chan — APPROVED** [review]

**Review by nawarani — APPROVED** [review]
The comments are descriptive and covers important bits. Especially the bitboard descriptions are very helpful.

**Pull request merged**